src/library/basetrackcache.cpp

```diff
@@ -532,10 +545,22 @@ void BaseTrackCache::filterAndSort(const QSet<TrackId>& trackIds,
         m_trackOrder.reserve(rows);
     }
 
-    while (query.next()) {
-        TrackId trackId(query.value(idColumn));
-        (*trackToIndex)[trackId] = m_trackOrder.size();
-        m_trackOrder.append(trackId);
+    if (sortedIndexes->isEmpty()) {
+        // Loop through tracks and store their indexes
+        // in the query result.
+        while (query.next()) {
+            TrackId trackId(query.value(idColumn));
+            (*trackToIndex)[trackId] = m_trackOrder.size();
+            m_trackOrder.append(trackId);
+        }
+    } else {
+        // Loop through tracks and store their indexes
+        // from the previous sorted order.
+        while (query.next()) {
```

Comment:
If sortedIndexes would be a vector we can just loop over it and assign the sorting value consecutive if found.

Reply:
> This PR doesn't work for me. Search results are not updated as expected when modifying the query.
> 
> I am also concerned about storing even more data in memory.

You are right, that it is a little bit problematic because more memory has to be used. Anyway, I couldn't find out another solution because it is necessary to store sorted tracks order. While tracks are chosen in filterAndSort() method it is necessary to have available the previous sorted order.

Reply:
> If sortedIndexes would be a vector we can just loop over it and assign the sorting value consecutive if found.

Ok, I'll try to remake it.

Reply:
Thank you. 

Some Ideas regarding the extra memory of redundant data:

I think we can safely reuse QVector<RowInfo> BaseSqlTableModel::m_rowInfo which holds the old sorting and is used to implement the QAbstractTableModel interface.

BaseTrackCache::m_trackOrder this was original a local variable from BaseTrackCache::filterAndSort() made a class member to keep the allocated memory for later calls, this has changed content due to the shared nature of BaseTrackCache. Maybe we can also get rid of this ...

Reply:
Thank you for the tips. Sounds good to me. I will try what will be possible asap. I agree, that adding this new feature can be a little bit problematic with wasting memory or data duplication in some ways.